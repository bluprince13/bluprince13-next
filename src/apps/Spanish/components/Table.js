import { styled } from '@mui/material/styles';

import TableColumn from './TableColumn'
import TableHeader from './TableHeader'
import TableRow from './TableRow'

import Arrow from './Arrow'

const PREFIX = 'Table';

const classes = {
    root: `${PREFIX}-root`
};

const Root = styled('div')((
    {
        theme
    }
) => ({
    [`&.${classes.root}`]: {
        width: '45%',
        display: 'inline-block',
        margin: theme.spacing(2)
    }
}));

export default function Table({
    tense,
    pronouns,
    forms,
    pointsInTime,
    lineInTime
}) {


    const tenseName = tense.name

    const getConjugation = ({ formName }) => {
        const form = tense.forms.find((f) => f.name === formName)
        const conjugation = pronouns.map((subject) => form.conjugation[subject])
        return conjugation
    }

    return (
        <Root className={classes.root}>
            <TableRow>
                <TableHeader type="title">{tenseName}</TableHeader>
                {forms.map((f) => (
                    <TableHeader key={f}>{f}</TableHeader>
                ))}
            </TableRow>
            <TableRow>
                <TableColumn items={pronouns} />
                {forms.map((f) => (
                    <TableColumn
                        key={f}
                        items={getConjugation({
                            formName: f
                        })}
                    />
                ))}
            </TableRow>
            <Arrow pointsInTime={pointsInTime} lineInTime={lineInTime} />
        </Root>
    );
}
