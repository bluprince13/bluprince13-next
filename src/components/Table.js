import Table from '@mui/material/Table'
import TableBody from '@mui/material/TableBody'
import TableCell, { tableCellClasses } from '@mui/material/TableCell'
import TableContainer from '@mui/material/TableContainer'
import TableHead from '@mui/material/TableHead'
import TableRow from '@mui/material/TableRow'
import Paper from '@mui/material/Paper'
import Typography from '@mui/material/Typography'
import { styled } from '@mui/material/styles'

const StyledTableCell = styled(TableCell)(({ theme }) => ({
    [`&.${tableCellClasses.head}`]: {
        backgroundColor: theme.palette.primary.light,
        color: theme.palette.common.white,
        fontSize: 16
    },
    [`&.${tableCellClasses.body}`]: {
        fontSize: 14
    }
}))

const StyledTableRow = styled(TableRow)(({ theme }) => ({
    '&:nth-of-type(odd)': {
        backgroundColor: theme.palette.action.hover
    },
    // hide last border
    '&:last-child td, &:last-child th': {
        border: 0
    }
}))

export default function BasicTable({ rows, columns }) {
    return (
        <TableContainer component={Paper}>
            <Table sx={{ minWidth: 650 }} aria-label="simple table">
                <TableHead>
                    <TableRow>
                        {columns.map(({ name }) => (
                            <StyledTableCell scope="col">
                                {name}
                            </StyledTableCell>
                        ))}
                    </TableRow>
                </TableHead>
                <TableBody>
                    {rows.map((row, row_index) => (
                        <StyledTableRow
                            key={row_index}
                            sx={{
                                '&:last-child td, &:last-child th': {
                                    border: 0
                                }
                            }}
                        >
                            {columns.map(({ key }, col_index) => (
                                <StyledTableCell
                                    key={key}
                                    component={col_index == 0 ? 'th' : null}
                                    scope={col_index == 0 ? 'row' : null}
                                >
                                    <Typography
                                        variant="body1"
                                        style={{ whiteSpace: 'pre-line' }}
                                    >
                                        {row[key]}
                                    </Typography>
                                </StyledTableCell>
                            ))}
                        </StyledTableRow>
                    ))}
                </TableBody>
            </Table>
        </TableContainer>
    )
}
