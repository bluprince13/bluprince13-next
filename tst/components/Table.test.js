import Table from '@Components/Table'
import { render } from '@testing-library/react'

const columns = [
    { name: 'col1name', key: 'col1' },
    { name: 'col2name', key: 'col2' }
]

const rows = [
    {
        col1: 'row1-col1',
        col2: 'row1-col2'
    },
    {
        col1: 'row2-col1',
        col2: 'row2-col2'
    }
]

describe('Table', () => {
    const setup = () => {
        const queries = render(<Table rows={rows} columns={columns} />)
        return queries
    }

    it('renders correctly', () => {
        const { asFragment } = setup()

        expect(asFragment()).toMatchSnapshot()
    })
})
