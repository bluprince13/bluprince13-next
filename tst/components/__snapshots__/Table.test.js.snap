// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
  >
    <table
      aria-label="simple table"
      class="MuiTable-root css-quj9j6-MuiTable-root"
    >
      <thead
        class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1obp5w9-MuiTableCell-root"
            scope="col"
          >
            col1name
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1obp5w9-MuiTableCell-root"
            scope="col"
          >
            col2name
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root css-1ffuvkz-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-ojbc7e-MuiTableCell-root"
            scope="row"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              style="white-space: pre-line;"
            >
              row1-col1
            </p>
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-ojbc7e-MuiTableCell-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              style="white-space: pre-line;"
            >
              row1-col2
            </p>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1ffuvkz-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-ojbc7e-MuiTableCell-root"
            scope="row"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              style="white-space: pre-line;"
            >
              row2-col1
            </p>
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-ojbc7e-MuiTableCell-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              style="white-space: pre-line;"
            >
              row2-col2
            </p>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
